lib/logstash/inputs/eventlog.rb

```diff
@@ -21,120 +22,62 @@ class LogStash::Inputs::EventLog < LogStash::Inputs::Base
+
+  # How frequently should tail check for new event logs in ms (default: 1 second)
+  config :interval, :validate => :number, :default => 1000
 
   public
   def register
 
     # wrap specified logfiles in suitable OR statements
-    @logfiles = @logfile.join("' OR TargetInstance.LogFile = '")
-
     @hostname = Socket.gethostname
-    @logger.info("Registering input eventlog://#{@hostname}/#{@logfile}")
+    @logger.info("Opening eventlog #{@logfile}")
 
-    if LogStash::Environment.jruby?
-      require "jruby-win32ole"
-    else
-      require "win32ole"
-    end
+    @eventlog = Win32::EventLog.open(@logfile)
   end # def register
 
   public
   def run(queue)
 
-    @wmi = WIN32OLE.connect("winmgmts://")
-    wmi_query = "Select * from __InstanceCreationEvent Where TargetInstance ISA 'Win32_NTLogEvent' And (TargetInstance.LogFile = '#{@logfiles}')"
-
     @logger.debug("Tailing Windows Event Log '#{@logfile}'")
 
-    begin
-      @events = @wmi.ExecNotificationQuery(wmi_query)
-    rescue => e
-      @logger.fatal("Unable to tail Windows Event Log: #{e.message}")
-      @logger.info("Windows Event Log Query: #{wmi_query}")
-      return # fatal scenario => exit
-    end
+    old_total = @eventlog.total_records()
+    flags     = Win32::EventLog::FORWARDS_READ | Win32::EventLog::SEEK_READ
+    rec_num   = @eventlog.read_last_event.record_number
 
     while !stop?
-
-      begin
-        # timeout is needed here otherwise NextEvent prevents logstash from exiting
-        notification = @events.NextEvent(1000) # 1000 ms
-      rescue Java::OrgRacobCom::ComFailException
-        next
-      end
-
-      event = notification.TargetInstance
-
-      timestamp = to_timestamp(event.TimeGenerated)
-
-      e = LogStash::Event.new(
-        "host" => @hostname,
-        "path" => @logfile,
-        "type" => @type,
-        LogStash::Event::TIMESTAMP => timestamp
-      )
-
-      %w{Category CategoryString ComputerName EventCode EventIdentifier
-          EventType Logfile Message RecordNumber SourceName
-          TimeGenerated TimeWritten Type User
-      }.each{
-          |property| e[property] = event.send property
-      }
-
-      if LogStash::Environment.jruby?
-        # unwrap jruby-win32ole racob data
-        e["InsertionStrings"] = unwrap_racob_variant_array(event.InsertionStrings)
-        data = unwrap_racob_variant_array(event.Data)
-        # Data is an array of signed shorts, so convert to bytes and pack a string
-        e["Data"] = data.map{|byte| (byte > 0) ? byte : 256 + byte}.pack("c*")
-      else
-        # win32-ole data does not need to be unwrapped
-        e["InsertionStrings"] = event.InsertionStrings
-        e["Data"] = event.Data
+      new_total = @eventlog.total_records()
+      if new_total != old_total
+        rec_num = @eventlog.oldest_record_number() if @eventlog.full?
+        @eventlog.read(flags, rec_num).each { |log| e = process(log); decorate(e); queue << e }
```

Comment:
I don't think so, but in case is happens we will be safe with a log message. I think that running with --debug can show us also if this happens, but just want to make sure Customers can see that something happened reading from the event log.